### ISSUE TITLE
Create a script + github action for rebaselining tests

### ISSUE THREAD

**Issue by sbc100** [emscripten-core/emscripten#23146]
We have a bunch of codesize tests that need to be rebaselined.  This need can occur under 2 different circumstances.  

1. A llvm or binaryen commit rolled in that changes the generated code.
2. An emscripten PR causes changes to the generated code.

I think it would be good if we can avoid mixing (1) and (2).  These means that the main branch should always be updated to include changes from llvm/binaryen before the changes for specific PR are landed (as part of that PR).

For (1) procedure I've been using looks something like this:

```
$ git checkout -b main upstream/main
$ ./emcc --clear-cache
$ test/runner other.*code_size* other.*codesize* --rebase
$ git add -u .
$ git commit -m "Rebaseline codesize expectations. NFC"
$ git show  # check everything looks reasonable
$ git push upsteam # direct push to upstream, no PR
```

There are several problems with this (as pointed out in https://github.com/emscripten-core/emscripten/pull/23126#discussion_r1879129835).

1. Its a lot of separate steps
2. It involves pushing directly to repo, without any review
3. Its hard to see (using `git diff`) exactly what effect the changes have.

I propose adding a new script that performs most of the step above that will have the following advantages:

1. It can analyze the results and present a human readable report (e.g. `test xxx regressed by 10 bytes (0.1%)`).
2. It can automatically create a PR with the report in the description.
3. It can be turned into a github action so it can be run on a remote machine (not a users machine where llvm and binaries version might not be exactly at `tot`).

**Comment by kripken** [comment]
Sounds great!

**Comment by brendandahl** [comment]
Yeah, sounds good. It's such a pain having the size tests randomly break and then having to figure out if it's my changes or some llvm/binaryen roll.

**Comment by sbc100** [comment]
We could even have a github action generate the code size changes for you and add to your PR?   That way you wouldn't need to make sure you have the exact `tot` version locally.

**Comment by sbc100** [comment]
For usage of the initial version of this script: https://github.com/emscripten-core/emscripten/pull/23188
